src/task/executable_task.rs

```diff
@@ -0,0 +1,379 @@
+        Ok(RunOutput {
+            exit_code: code,
+            stdout: stdout_handle.await.unwrap(),
+            stderr: stderr_handle.await.unwrap(),
+        })
+    }
+}
+
+/// A helper object that implements [`Display`] to display (with ascii color) the command of the
+/// task.
+struct ExecutableTaskConsoleDisplay<'p, 't> {
+    task: &'t ExecutableTask<'p>,
+}
+
+impl<'p, 't> Display for ExecutableTaskConsoleDisplay<'p, 't> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let command = self.task.task.as_single_command();
+        write!(
+            f,
+            "{}",
+            console::style(command.as_deref().unwrap_or("<alias>"))
+                .blue()
+                .bold()
+        )?;
+        if !self.task.additional_args.is_empty() {
+            write!(
+                f,
+                " {}",
+                console::style(self.task.additional_args.join(" ")).blue()
+            )?;
+        }
+        Ok(())
+    }
+}
+/// Helper function to create a pipe that we can get the output from.
+fn get_output_writer_and_handle() -> (ShellPipeWriter, JoinHandle<String>) {
+    let (reader, writer) = pipe();
+    let handle = reader.pipe_to_string_handle();
+    (writer, handle)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::path::Path;
+
+    #[tokio::test]
+    async fn test_ordered_commands() {
+        let file_content = r#"
+        [project]
+        name = "pixi"
+        channels = ["conda-forge"]
+        platforms = ["linux-64"]
+        [tasks]
+        root = "echo root"
+        task1 = {cmd="echo task1", depends_on=["root"]}
+        task2 = {cmd="echo task2", depends_on=["root"]}
+        top = {cmd="echo top", depends_on=["task1","task2"]}
+    "#;
+        let project = Project::from_manifest_str(Path::new(""), file_content.to_string()).unwrap();
+
+        let executable_tasks = ExecutableTask::from_cmd_args(
+            &project,
+            vec!["top".to_string(), "--test".to_string()],
+            Some(Platform::current()),
+        )
+        .get_ordered_dependencies()
+        .await
+        .unwrap();
+
+        let ordered_task_names: Vec<_> = executable_tasks
+            .iter()
+            .map(|task| task.task().as_single_command().unwrap())
+            .collect();
+
+        assert_eq!(
+            ordered_task_names,
+            vec!["echo root", "echo task1", "echo task2", "echo top"]
+        );
+
+        // Also check if the arguments are passed correctly
+        assert_eq!(
+            executable_tasks.last().unwrap().additional_args(),
+            vec!["--test".to_string()]
+        );
+    }
+
+    #[tokio::test]
+    async fn test_cycle_ordered_commands() {
+        let file_content = r#"
+        [project]
+        name = "pixi"
+        channels = ["conda-forge"]
+        platforms = ["linux-64"]
+        [tasks]
+        root = {cmd="echo root", depends_on=["task1"]}
+        task1 = {cmd="echo task1", depends_on=["root"]}
```

Comment:
I'm now thinking, shouldn't we error or warn on this? To avoid unwanted overseen behavior